### DISCUSSION TITLE
Stats API returning different values to dashboard

### DISCUSSION THREAD

**Discussion by lloydjatkinson** [plausible/analytics#2478] [Anything Else]
Hi,

The stats API is returning lower values than the dashboard. Compare the following example API calls to the dashboard for these pages. I am using the date range trick (from the start, to current date) to get all the statistics.

```
https://plausible.io/api/v1/stats/aggregate?site_id=lloydatkinson.net&period=custom&metrics=pageviews&date=2021-01-01,2022-11-29&filters=event:page==/posts/2022/default-exports-in-javascript-modules-are-terrible/

{
    "results": {
        "pageviews": {
            "value": 25752
        }
    }
}
```
![image](https://user-images.githubusercontent.com/5633938/204411826-88f138ee-b7b2-475e-b80a-c212dc52aaf1.png)

And then:

```
https://plausible.io/api/v1/stats/aggregate?site_id=lloydatkinson.net&period=custom&metrics=pageviews&date=2021-01-01,2022-11-29&filters=event:page==/posts/2022/modelling-workflows-with-finite-state-machines-in-dotnet/

{
    "results": {
        "pageviews": {
            "value": 12883
        }
    }
}
```
![image](https://user-images.githubusercontent.com/5633938/204411898-bcfb6a34-6511-488e-8fe0-de8f2dee0474.png)

**Comment by ukutaht** [comment]
@lloydjatkinson Thanks for reporting.

I believe the difference comes from using `entry_page` filter in one case and `page` filter in the other. In the UI these are not tagged by in the API they are named `session:entry_page` and `event:page` respectively.

When filtering by `visit:entry_page`, the pageviews number shows the **total number of combined pageviews from visits that entered through /posts/2022/modelling-workflows-with-finite-state-machines-in-dotnet/**. If someone landed on that blog post and then continued to navigate through your website, these are also counted in the 'pageviews' report.

However, when using the `event:page` filter, the pageviews number only counts **pageviews that happned on that page directly**.

I know this can be confusing. This difference in behaviour is the biggest problem we have with information architecture at the moment, and we're doing a lot of thinking internally how to make it more intuitive.

Any suggestions?

To understand the motivation better: when you filter by some other visit parameter, like say `visit:device=Desktop`, it would make sense that you want to see all pageviews from visits that have that attribute. `entry_page` is stored as a parameter on the visitor session so it acts the same way. I suppose we could special-case the `entry_page` and `exit_page` filters to make sure they only return pageviews that happened on that page directly.

**Reply by lloydjatkinson** [reply]
I see, I think I understand, mostly 😊

> Any suggestions?

I think some way of getting total page views as in **times the page has been viewed regardless of if it's an entry page or not** would be really helpful, so that the API reports the same numbers as the dashboard. Perhaps a `/total` endpoint? Afterall the current one is `/aggregate`.

I am finding the terminology a bit confusing when trying to use the dashboard at the same time as the API. Is there anyway currently that I'd be able to find out the page count, regardless of if it was an entry page or not, for specific pages? If so, what would the URL look like for the API call?

Essentially imagine a post, and you want to know how many people in total have ever read it.

**Reply by ukutaht** [reply]
Yes, this query you already use will give you the pageviews on that page specifically:

https://plausible.io/api/v1/stats/aggregate?site_id=lloydatkinson.net&period=custom&metrics=pageviews&date=2021-01-01,2022-11-29&filters=event:page==/posts/2022/modelling-workflows-with-finite-state-machines-in-dotnet/

This should match the number when you add `page=/posts/2022/modelling-workflows-with-finite-state-machines-in-dotnet/` filter on the dashboard. The difference you see in the UI is because you were using `entry_page=/posts/2022/modelling-workflows-with-finite-state-machines-in-dotnet/` filter instead (at least in the screenshot).

In the UI, make sure you're using the 'Top Pages' report instead of 'Entry pages' when clicking on a page:

<img width="591" alt="Screenshot 2022-11-30 at 11 39 02" src="https://user-images.githubusercontent.com/3731516/204761177-e0aa6cf1-9469-47f1-b226-902c2e0ebd9b.png">

BTW: We are redesigning this part of the API and UI to be clearer. We don't want to make any breaking changes at this point to the API but we'll see what we can do for now.

**Reply by lloydjatkinson** [reply]
I see, thank you for explaining!
